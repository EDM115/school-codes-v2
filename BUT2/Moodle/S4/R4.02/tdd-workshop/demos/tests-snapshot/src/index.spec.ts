import { expect, it } from 'vitest';
import { starshipsAPI, toUpperCase } from '.';

it('toUpperCase', () => {
  const result = toUpperCase('foobar');
  expect(result).toMatchSnapshot();
})

it('starship API', async () => {
  const result = await starshipsAPI(1);
  expect(result).toMatchSnapshot();
})