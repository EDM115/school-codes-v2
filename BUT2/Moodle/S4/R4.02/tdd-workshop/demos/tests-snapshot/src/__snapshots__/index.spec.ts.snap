// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`starship API 1`] = `
{
  "count": 36,
  "next": "https://swapi.dev/api/starships/?page=2",
  "previous": null,
  "results": [
    {
      "MGLT": "60",
      "cargo_capacity": "3000000",
      "consumables": "1 year",
      "cost_in_credits": "3500000",
      "created": "2014-12-10T14:20:33.369000Z",
      "crew": "30-165",
      "edited": "2014-12-20T21:23:49.867000Z",
      "films": [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/3/",
        "https://swapi.dev/api/films/6/",
      ],
      "hyperdrive_rating": "2.0",
      "length": "150",
      "manufacturer": "Corellian Engineering Corporation",
      "max_atmosphering_speed": "950",
      "model": "CR90 corvette",
      "name": "CR90 corvette",
      "passengers": "600",
      "pilots": [],
      "starship_class": "corvette",
      "url": "https://swapi.dev/api/starships/2/",
    },
    {
      "MGLT": "60",
      "cargo_capacity": "36000000",
      "consumables": "2 years",
      "cost_in_credits": "150000000",
      "created": "2014-12-10T15:08:19.848000Z",
      "crew": "47,060",
      "edited": "2014-12-20T21:23:49.870000Z",
      "films": [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
      ],
      "hyperdrive_rating": "2.0",
      "length": "1,600",
      "manufacturer": "Kuat Drive Yards",
      "max_atmosphering_speed": "975",
      "model": "Imperial I-class Star Destroyer",
      "name": "Star Destroyer",
      "passengers": "n/a",
      "pilots": [],
      "starship_class": "Star Destroyer",
      "url": "https://swapi.dev/api/starships/3/",
    },
    {
      "MGLT": "70",
      "cargo_capacity": "180000",
      "consumables": "1 month",
      "cost_in_credits": "240000",
      "created": "2014-12-10T15:48:00.586000Z",
      "crew": "5",
      "edited": "2014-12-20T21:23:49.873000Z",
      "films": [
        "https://swapi.dev/api/films/1/",
      ],
      "hyperdrive_rating": "1.0",
      "length": "38",
      "manufacturer": "Sienar Fleet Systems, Cyngus Spaceworks",
      "max_atmosphering_speed": "1000",
      "model": "Sentinel-class landing craft",
      "name": "Sentinel-class landing craft",
      "passengers": "75",
      "pilots": [],
      "starship_class": "landing craft",
      "url": "https://swapi.dev/api/starships/5/",
    },
    {
      "MGLT": "10",
      "cargo_capacity": "1000000000000",
      "consumables": "3 years",
      "cost_in_credits": "1000000000000",
      "created": "2014-12-10T16:36:50.509000Z",
      "crew": "342,953",
      "edited": "2014-12-20T21:26:24.783000Z",
      "films": [
        "https://swapi.dev/api/films/1/",
      ],
      "hyperdrive_rating": "4.0",
      "length": "120000",
      "manufacturer": "Imperial Department of Military Research, Sienar Fleet Systems",
      "max_atmosphering_speed": "n/a",
      "model": "DS-1 Orbital Battle Station",
      "name": "Death Star",
      "passengers": "843,342",
      "pilots": [],
      "starship_class": "Deep Space Mobile Battlestation",
      "url": "https://swapi.dev/api/starships/9/",
    },
    {
      "MGLT": "75",
      "cargo_capacity": "100000",
      "consumables": "2 months",
      "cost_in_credits": "100000",
      "created": "2014-12-10T16:59:45.094000Z",
      "crew": "4",
      "edited": "2014-12-20T21:23:49.880000Z",
      "films": [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
      ],
      "hyperdrive_rating": "0.5",
      "length": "34.37",
      "manufacturer": "Corellian Engineering Corporation",
      "max_atmosphering_speed": "1050",
      "model": "YT-1300 light freighter",
      "name": "Millennium Falcon",
      "passengers": "6",
      "pilots": [
        "https://swapi.dev/api/people/13/",
        "https://swapi.dev/api/people/14/",
        "https://swapi.dev/api/people/25/",
        "https://swapi.dev/api/people/31/",
      ],
      "starship_class": "Light freighter",
      "url": "https://swapi.dev/api/starships/10/",
    },
    {
      "MGLT": "80",
      "cargo_capacity": "110",
      "consumables": "1 week",
      "cost_in_credits": "134999",
      "created": "2014-12-12T11:00:39.817000Z",
      "crew": "2",
      "edited": "2014-12-20T21:23:49.883000Z",
      "films": [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
      ],
      "hyperdrive_rating": "1.0",
      "length": "14",
      "manufacturer": "Koensayr Manufacturing",
      "max_atmosphering_speed": "1000km",
      "model": "BTL Y-wing",
      "name": "Y-wing",
      "passengers": "0",
      "pilots": [],
      "starship_class": "assault starfighter",
      "url": "https://swapi.dev/api/starships/11/",
    },
    {
      "MGLT": "100",
      "cargo_capacity": "110",
      "consumables": "1 week",
      "cost_in_credits": "149999",
      "created": "2014-12-12T11:19:05.340000Z",
      "crew": "1",
      "edited": "2014-12-20T21:23:49.886000Z",
      "films": [
        "https://swapi.dev/api/films/1/",
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
      ],
      "hyperdrive_rating": "1.0",
      "length": "12.5",
      "manufacturer": "Incom Corporation",
      "max_atmosphering_speed": "1050",
      "model": "T-65 X-wing",
      "name": "X-wing",
      "passengers": "0",
      "pilots": [
        "https://swapi.dev/api/people/1/",
        "https://swapi.dev/api/people/9/",
        "https://swapi.dev/api/people/18/",
        "https://swapi.dev/api/people/19/",
      ],
      "starship_class": "Starfighter",
      "url": "https://swapi.dev/api/starships/12/",
    },
    {
      "MGLT": "105",
      "cargo_capacity": "150",
      "consumables": "5 days",
      "cost_in_credits": "unknown",
      "created": "2014-12-12T11:21:32.991000Z",
      "crew": "1",
      "edited": "2014-12-20T21:23:49.889000Z",
      "films": [
        "https://swapi.dev/api/films/1/",
      ],
      "hyperdrive_rating": "1.0",
      "length": "9.2",
      "manufacturer": "Sienar Fleet Systems",
      "max_atmosphering_speed": "1200",
      "model": "Twin Ion Engine Advanced x1",
      "name": "TIE Advanced x1",
      "passengers": "0",
      "pilots": [
        "https://swapi.dev/api/people/4/",
      ],
      "starship_class": "Starfighter",
      "url": "https://swapi.dev/api/starships/13/",
    },
    {
      "MGLT": "40",
      "cargo_capacity": "250000000",
      "consumables": "6 years",
      "cost_in_credits": "1143350000",
      "created": "2014-12-15T12:31:42.547000Z",
      "crew": "279,144",
      "edited": "2014-12-20T21:23:49.893000Z",
      "films": [
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
      ],
      "hyperdrive_rating": "2.0",
      "length": "19000",
      "manufacturer": "Kuat Drive Yards, Fondor Shipyards",
      "max_atmosphering_speed": "n/a",
      "model": "Executor-class star dreadnought",
      "name": "Executor",
      "passengers": "38000",
      "pilots": [],
      "starship_class": "Star dreadnought",
      "url": "https://swapi.dev/api/starships/15/",
    },
    {
      "MGLT": "20",
      "cargo_capacity": "19000000",
      "consumables": "6 months",
      "cost_in_credits": "unknown",
      "created": "2014-12-15T12:34:52.264000Z",
      "crew": "6",
      "edited": "2014-12-20T21:23:49.895000Z",
      "films": [
        "https://swapi.dev/api/films/2/",
        "https://swapi.dev/api/films/3/",
      ],
      "hyperdrive_rating": "4.0",
      "length": "90",
      "manufacturer": "Gallofree Yards, Inc.",
      "max_atmosphering_speed": "650",
      "model": "GR-75 medium transport",
      "name": "Rebel transport",
      "passengers": "90",
      "pilots": [],
      "starship_class": "Medium transport",
      "url": "https://swapi.dev/api/starships/17/",
    },
  ],
}
`;

exports[`toUpperCase 1`] = `"FOOBAR"`;
